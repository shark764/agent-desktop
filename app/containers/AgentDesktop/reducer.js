/*
 *
 * AgentDesktop reducer
 *
 */

import { fromJS } from 'immutable';
import {
  SET_PRESENCE,
  SET_INTERACTION_STATUS,
  ADD_INTERACTION,
  REMOVE_INTERACTION,
  SET_MESSAGE_HISTORY,
  ASSIGN_CONTACT,
  ADD_MESSAGE,
  SELECT_INTERACTION,
  SET_CUSTOM_FIELDS,
  EMAIL_CREATE_REPLY,
  EMAIL_CANCEL_REPLY,
} from './constants';

const initialState = fromJS({
  interactions: [
    // XXX uncomment below to mock email(s)
    // {
    //   channelType: 'email',
    //   interactionId: '0000000-0000-0000-0000-111111111',
    //   status: 'work-accepted', // 'work-offer',
    //   timeout: new Date(Date.now() + 60000).toISOString(),
    //   email: {
    //     to: 'support@help.com',
    //     from: 'j.englebert@yahoo.com',
    //     timestamp: new Date().toISOString(),
    //     subject: 'Files not uploading to my Cloud account',
    //     content: 'Hello,<br/><br/>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.<br/><br/><b>John Englebert</b><br/>Software Developer<br/>An Organization<br/>313.218.9814',
    //   },
    // },
    // {
    //   channelType: 'email',
    //   interactionId: '0000000-0000-0000-0000-222222222222',
    //   status: 'work-accepted', // 'work-offer',
    //   timeout: new Date(Date.now() + 60000).toISOString(),
    //   email: {
    //     to: 'support@help.com',
    //     from: 'test@yahoo.com',
    //     timestamp: new Date().toISOString(),
    //     subject: 'Files not uploading to my Cloud account',
    //     content: 'Hello,<br/><br/>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.<br/><br/><b>John Englebert</b><br/>Software Developer<br/>An Organization<br/>313.218.9814',
    //   },
    // },

    // XXX uncomment below to mock voice interaction
    // {
    //   channelType: 'voice',
    //   interactionId: '0000000-0000-0000-0000-3333333333333',
    //   status: 'work-accepted', // 'work-offer',
    //   timeout: new Date(Date.now() + 60000).toISOString(),
    //   number: '313.412.6623',
    // },
  ],
});

function agentDesktopReducer(state = initialState, action) {
  switch (action.type) {
    case SET_PRESENCE:
      return state
        .set('presence', action.response.state)
        .set('availablePresences', fromJS(action.response.availableStates));
    case SET_INTERACTION_STATUS: {
      const interactionIndex = state.get('interactions').findIndex(
        (interaction) => interaction.get('interactionId') === action.interactionId
      );
      if (interactionIndex !== -1) {
        const automaticallyAcceptInteraction = action.newStatus === 'work-accepting' && state.get('selectedInteractionId') === undefined;
        return state
          .update('interactions',
            (interactions) =>
              interactions.update(
                interactionIndex,
                (interaction) => interaction.set('status', action.newStatus)
              )
          ).set('selectedInteractionId',
            automaticallyAcceptInteraction
            ? action.interactionId
            : state.get('selectedInteractionId'));
      } else {
        return state;
      }
    }
    case ADD_INTERACTION: {
      const interaction = {
        channelType: action.response.channelType,
        customerAvatarIndex: Math.floor(Math.random() * 17),
        interactionId: action.response.interactionId,
        status: 'work-offer',
        timeout: action.response.timeout,
      };
      return state
        .set('interactions', state.get('interactions').push(fromJS(interaction)));
    }
    case REMOVE_INTERACTION:
      return state
        .set('interactions', state.get('interactions').filterNot((interaction) =>
          interaction.get('interactionId') === action.interactionId
        ))
        .set('selectedInteractionId', state.get('selectedInteractionId') === action.interactionId ? undefined : state.get('selectedInteractionId'));
    case SET_MESSAGE_HISTORY: {
      if (action.response && action.response.length > 0) {
        const interactionIndex = state.get('interactions').findIndex(
          (interaction) => interaction.get('interactionId') === action.response[0].channelId
        );
        if (interactionIndex !== -1) {
          const messageHistoryItems = action.response.map((messageHistoryItem) => ({
            text: messageHistoryItem.payload.body.text,
            from: messageHistoryItem.payload.metadata && messageHistoryItem.payload.metadata.name ? messageHistoryItem.payload.metadata.name : messageHistoryItem.payload.from,
            type: messageHistoryItem.payload.metadata ? messageHistoryItem.payload.metadata.type : messageHistoryItem.payload.type,
            timestamp: messageHistoryItem.payload.timestamp,
          }));
          return state
            .update('interactions',
              (interactions) =>
                interactions.update(
                  interactionIndex,
                  (interaction) => interaction.set('messageHistory', fromJS(messageHistoryItems))
                )
            );
        } else {
          return state;
        }
      } else {
        return state;
      }
    }
    case ASSIGN_CONTACT: {
      const interactionIndex = state.get('interactions').findIndex(
        (interaction) => interaction.get('interactionId') === action.interactionId
      );
      if (interactionIndex !== -1) {
        return state
          .update('interactions',
            (interactions) =>
              interactions.update(
                interactionIndex,
                (interaction) => interaction.set('contact', fromJS(action.contact))
              )
          );
      } else {
        return state;
      }
    }
    case ADD_MESSAGE: {
      const interactionIndex = state.get('interactions').findIndex(
        (interaction) => interaction.get('interactionId') === action.response.to
      );
      if (interactionIndex !== -1) {
        const message = action.response;
        const messageHistoryItem = {
          text: message.body.text,
          from: message.metadata && message.metadata.name ? message.metadata.name : message.from,
          type: message.metadata.type,
          timestamp: message.timestamp,
        };
        return state
          .update('interactions',
          (interactions) =>
            interactions.update(
              interactionIndex,
              (interaction) => interaction.update('messageHistory', (messageHistory) => messageHistory.push(fromJS(messageHistoryItem)))
                .set('hasUnreadMessage', state.get('selectedInteractionId') !== interaction.get('interactionId'))
          )
        );
      } else {
        return state;
      }
    }
    case SELECT_INTERACTION: {
      const interactionIndex = state.get('interactions').findIndex(
        (interaction) => interaction.get('interactionId') === action.interactionId
      );
      if (interactionIndex !== -1) {
        return state
          .set('selectedInteractionId', action.interactionId)
          .update('interactions',
            (interactions) =>
              interactions.update(
                interactionIndex,
                (interaction) => interaction.set('hasUnreadMessage', false)
              )
          );
      } else {
        return state;
      }
    }
    case SET_CUSTOM_FIELDS: {
      const interactionIndex = state.get('interactions').findIndex(
        (interaction) => interaction.get('interactionId') === action.interactionId
      );
      if (interactionIndex !== -1) {
        return state
          .update('interactions',
            (interactions) =>
              interactions.update(
                interactionIndex,
                (interaction) => interaction.set('customFields', action.customFields)
              )
          );
      } else {
        return state;
      }
    }
    case EMAIL_CREATE_REPLY: {
      const interactionIndex = state.get('interactions').findIndex(
        (interaction) => interaction.get('interactionId') === action.interactionId
      );
      if (interactionIndex !== -1 && state.get('interactions').get(interactionIndex).get('channelType') === 'email') {
        return state
          .update('interactions',
            (interactions) =>
              interactions.update(
                interactionIndex,
                (interaction) => interaction.set('email', interaction.get('email')
                  .set('reply', fromJS({
                    to: interaction.get('email').get('from'),
                    subject: `RE: ${interaction.get('email').get('subject')}`,
                  }))
                )
              )
          );
      } else {
        return state;
      }
    }
    case EMAIL_CANCEL_REPLY: {
      const interactionIndex = state.get('interactions').findIndex(
        (interaction) => interaction.get('interactionId') === action.interactionId
      );
      if (interactionIndex !== -1 && state.get('interactions').get(interactionIndex).get('channelType') === 'email') {
        return state
          .update('interactions',
            (interactions) =>
              interactions.update(
                interactionIndex,
                (interaction) => interaction.set('email', interaction.get('email')
                  .set('reply', undefined)
                )
              )
          );
      } else {
        return state;
      }
    }
    default:
      return state;
  }
}

export default agentDesktopReducer;
