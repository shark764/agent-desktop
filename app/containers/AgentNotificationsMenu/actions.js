/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 *
 * AgentNotificationsMenu actions
 *
 */

import * as ACTIONS from './constants';

// Reducer actions

export function setAudioNotificationsPreference(audioNotifications) {
  return {
    type: ACTIONS.SET_AUDIO_NOTIFICATIONS_PREFERENCE,
    audioNotifications,
  };
}

export function setVisualNotificationsPreference(visualNotifications) {
  return {
    type: ACTIONS.SET_VISUAL_NOTIFICATIONS_PREFERENCE,
    visualNotifications,
  };
}

// Saga actions

export function initializeNotificatonPreferences() {
  return {
    type: ACTIONS.INITIALIZE_NOTIFICATION_PREFERENCES,
  };
}

export function toggleAudioNotificationsPreference() {
  return {
    type: ACTIONS.TOGGLE_AUDIO_NOTIFICATIONS_PREFERENCE,
  };
}

export function toggleVisualNotificationsPreference() {
  return {
    type: ACTIONS.TOGGLE_VISUAL_NOTIFICATIONS_PREFERENCE,
  };
}
