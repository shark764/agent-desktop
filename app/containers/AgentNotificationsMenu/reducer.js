/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 *
 * AgentNotificationsMenu reducer
 *
 */

import { fromJS } from 'immutable';
import * as ACTIONS from './constants';

const initialState = fromJS({
  audioNotifications: true,
  visualNotifications: true,
});

export default function notificationPreferencesReducer(
  state = initialState,
  action
) {
  switch (action.type) {
    case ACTIONS.SET_AUDIO_NOTIFICATIONS_PREFERENCE:
      return state.set('audioNotifications', action.audioNotifications);
    case ACTIONS.SET_VISUAL_NOTIFICATIONS_PREFERENCE:
      return state.set('visualNotifications', action.visualNotifications);
    default:
      return state;
  }
}
