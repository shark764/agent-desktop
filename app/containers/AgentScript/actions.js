/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 *
 * AgentScript actions
 *
 */

import { SEND_SCRIPT } from './constants';

export function sendScript(interactionId, script, dismissed) {
  return {
    type: SEND_SCRIPT,
    interactionId,
    script,
    dismissed,
  };
}
