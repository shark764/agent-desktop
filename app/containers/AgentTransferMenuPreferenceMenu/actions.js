/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 *
 * AgentTransferMenuPreferenceMenu actions
 *
 */

import * as ACTIONS from './constants';

// Reducer actions
export function setAgentsTransferMenuPreference(agentsTransferMenu) {
  return {
    type: ACTIONS.SET_AGENTS_TRANSFER_MENU_PREFERENCE,
    agentsTransferMenu,
  };
}

//  Queues Reducer Actions
export function setShowQueuesTransferMenuPreference(showQueues) {
  return {
    type: ACTIONS.SET_SHOW_QUEUES_TRANSFER_MENU_PREFERENCE,
    showQueues,
  };
}

export function toggleSelectedQueueTransferMenuPreferenceOnState(queue) {
  return {
    type: ACTIONS.TOGGLE_SELECTED_QUEUE_TRANSFER_MENU_PREFERENCE_ON_STATE,
    queue,
  };
}

export function toggleAllSelectedQueuesTransferMenuPreferenceOnState(queues) {
  return {
    type: ACTIONS.TOGGLE_ALL_SELECTED_QUEUES_TRANSFER_MENU_PREFERENCE_ON_STATE,
    queues,
  };
}

export function toggleSelectedQueues(queues) {
  return {
    type: ACTIONS.TOGGLE_SELECTED_QUEUES_TRANSFER_MENU_PREFERENCE,
    queues,
  };
}

//  Transfer Lists Reducer Actions
export function setShowTransferListTransferMenuPreference(showTransferLists) {
  return {
    type: ACTIONS.SET_SHOW_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE,
    showTransferLists,
  };
}

export function toggleSelectedTransferListTransferMenuPreferenceOnState(
  transferList
) {
  return {
    type:
      ACTIONS.TOGGLE_SELECTED_TRANSFER_LIST_TRANSFER_MENU_PREFERENCE_ON_STATE,
    transferList,
  };
}

export function toggleAllSelectedTransferListsTransferMenuPreferenceOnState(
  transferLists
) {
  return {
    type:
      ACTIONS.TOGGLE_ALL_SELECTED_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE_ON_STATE,
    transferLists,
  };
}

export function toggleSelectedTransferLists(transferLists) {
  return {
    type: ACTIONS.TOGGLE_SELECTED_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE,
    transferLists,
  };
}

// Saga actions
export function initializeTransferMenuPreferences() {
  return {
    type: ACTIONS.INITIALIZE_TRANSFER_MENU_PREFERENCES,
  };
}

export function toggleAgentsTransferMenuPreference() {
  return {
    type: ACTIONS.TOGGLE_AGENTS_TRANSFER_MENU_PREFERENCE,
  };
}

export function toggleShowQueuesTransferMenuPreference() {
  return {
    type: ACTIONS.TOGGLE_SHOW_QUEUES_TRANSFER_MENU_PREFERENCE,
  };
}

export function toggleShowTransferListsTransferMenuPreference() {
  return {
    type: ACTIONS.TOGGLE_SHOW_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE,
  };
}

export function setPreferenceMenuQueuesLoading(preferenceMenuQueuesLoading) {
  return {
    type: ACTIONS.SET_PREFERENCE_MENU_QUEUES_LOADING,
    preferenceMenuQueuesLoading,
  };
}

export function setPreferenceMenuTransferListsLoading(
  preferenceMenuTransferListsLoading
) {
  return {
    type: ACTIONS.SET_PREFERENCE_MENU_TRANSFER_LISTS_LOADING,
    preferenceMenuTransferListsLoading,
  };
}

export function toggleSelectedQueueTransferMenuPreference(queue) {
  return {
    type: ACTIONS.TOGGLE_SELECTED_QUEUE_TRANSFER_MENU_PREFERENCE,
    queue,
  };
}

export function toggleAllSelectedQueuesTransferMenuPreference(queues) {
  return {
    type: ACTIONS.TOGGLE_ALL_SELECTED_QUEUES_TRANSFER_MENU_PREFERENCE,
    queues,
  };
}

export function toggleSelectedTransferListTransferMenuPreference(transferList) {
  return {
    type: ACTIONS.TOGGLE_SELECTED_TRANSFER_LIST_TRANSFER_MENU_PREFERENCE,
    transferList,
  };
}

export function toggleAllSelectedTransferListsTransferMenuPreference(
  transferLists
) {
  return {
    type: ACTIONS.TOGGLE_ALL_SELECTED_TRANSFER_LISTS_TRANSFER_MENU_PREFERENCE,
    transferLists,
  };
}
