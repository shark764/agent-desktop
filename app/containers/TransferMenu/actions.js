import * as ACTIONS from './constants';

// REDUCER ACTIONS:

export function setResourceCapactiy(resourceCapacity) {
  return {
    type: ACTIONS.SET_RESOURCE_CAPACITY,
    resourceCapacity,
  };
}

export function setUsers(users) {
  return {
    type: ACTIONS.SET_USERS,
    users,
  };
}

export function getAndSetTransferLists(transferLists) {
  return {
    type: ACTIONS.GET_AND_SET_TRANSFER_LISTS,
    transferLists,
  };
}

export function setQueuesListVisibleState(queuesListVisibleState) {
  return {
    type: ACTIONS.SET_QUEUES_LIST_VISIBLE_STATE,
    queuesListVisibleState,
  };
}

export function setAgentsListVisibleState(agentsListVisibleState) {
  return {
    type: ACTIONS.SET_AGENTS_LIST_VISIBLE_STATE,
    agentsListVisibleState,
  };
}

export function setTransferListsVisibleState(transferListsVisibleState) {
  return {
    type: ACTIONS.SET_TRANSFER_LIST_VISIBLE_STATE,
    transferListsVisibleState,
  };
}

export function setTransferSearchInput(transferSearchInput) {
  return {
    type: ACTIONS.SET_TRANSFER_SEARCH_INPUT,
    transferSearchInput,
  };
}

export function setFocusedTransferItemIndex(focusedTransferItemIndex) {
  return {
    type: ACTIONS.SET_FOCUSED_TRANSFER_ITEM_INDEX,
    focusedTransferItemIndex,
  };
}

export function setTransferTabIndex(transferTabIndex) {
  return {
    type: ACTIONS.SET_TRANSFER_TAB_INDEX,
    transferTabIndex,
  };
}

export function setShowTransferDialPad(showTransferDialpad) {
  return {
    type: ACTIONS.SET_SHOW_TRANSFER_DIAL_PAD,
    showTransferDialpad,
  };
}

// Saga Actions:

export function setTransferLists() {
  return {
    type: ACTIONS.SET_TRANSFER_LISTS,
  };
}
export function initializeQueuesAgentsVisibleState() {
  return {
    type: ACTIONS.SET_INITIAL_TRANSFER_MENUS_VISIBLE_STATE,
  };
}

export function updateQueuesListVisibleState() {
  return {
    type: ACTIONS.UPDATE_QUEUES_LIST_VISIBLE_STATE,
  };
}

export function updateAgentsListVisibleState() {
  return {
    type: ACTIONS.UPDATE_AGENTS_LIST_VISIBLE_STATE,
  };
}

export function updateTransferListVisibleState(transferListId) {
  return {
    type: ACTIONS.UPDATE_TRANSFER_LIST_VISIBLE_STATE,
    transferListId,
  };
}
export function tearDownTransferMenuStates() {
  return {
    type: ACTIONS.TEAR_DOWN_TRANSFER_MENU_STATES,
  };
}
