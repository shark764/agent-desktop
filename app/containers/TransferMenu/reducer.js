/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

/*
 *
 * Toolbar reducer
 *
 */

import { fromJS } from 'immutable';
import * as ACTIONS from './constants';

const initialState = fromJS({
  resourceCapacity: undefined,
  users: undefined,
  transferLists: ['loading'],
  queuesListVisibleState: undefined,
  agentsListVisibleState: undefined,
  transferListsVisibleState: {},
  transferSearchInput: '',
  transferTabIndex: 0,
  focusedTransferItemIndex: -1,
  showTransferDialpad: false,
});

export default function(state = initialState, action) {
  switch (action.type) {
    case ACTIONS.SET_RESOURCE_CAPACITY: {
      return state.set('resourceCapacity', fromJS(action.resourceCapacity));
    }
    case ACTIONS.SET_USERS: {
      return state.set('users', fromJS(action.users));
    }
    case ACTIONS.GET_AND_SET_TRANSFER_LISTS: {
      return state.set('transferLists', fromJS(action.transferLists));
    }
    case ACTIONS.SET_QUEUES_LIST_VISIBLE_STATE: {
      return state.set('queuesListVisibleState', action.queuesListVisibleState);
    }
    case ACTIONS.SET_AGENTS_LIST_VISIBLE_STATE: {
      return state.set('agentsListVisibleState', action.agentsListVisibleState);
    }
    case ACTIONS.SET_TRANSFER_LIST_VISIBLE_STATE: {
      return state.set(
        'transferListsVisibleState',
        fromJS(action.transferListsVisibleState)
      );
    }
    case ACTIONS.SET_TRANSFER_SEARCH_INPUT: {
      return state.set('transferSearchInput', action.transferSearchInput);
    }
    case ACTIONS.SET_TRANSFER_TAB_INDEX: {
      return state.set('transferTabIndex', action.transferTabIndex);
    }
    case ACTIONS.SET_FOCUSED_TRANSFER_ITEM_INDEX: {
      return state.set(
        'focusedTransferItemIndex',
        action.focusedTransferItemIndex
      );
    }
    case ACTIONS.SET_SHOW_TRANSFER_DIAL_PAD: {
      return state.set('showTransferDialpad', action.showTransferDialpad);
    }
    default:
      return state;
  }
}
