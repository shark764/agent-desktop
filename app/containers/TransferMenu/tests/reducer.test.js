import { Map } from 'immutable';
import reducer from '../reducer';
import * as action from '../actions';

describe('TransferMenu reducer tests', () => {
  it('returns initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });
  it('sets queuesListVisibleState', () => {
    expect(
      reducer(Map(), action.setQueuesListVisibleState(true))
    ).toMatchSnapshot();
  });
  it('sets agentsListVisibleState', () => {
    expect(
      reducer(Map(), action.setAgentsListVisibleState(true))
    ).toMatchSnapshot();
  });
  it('updates showTransferDialpad state', () => {
    expect(
      reducer(Map(), action.setShowTransferDialPad(true))
    ).toMatchSnapshot();
  });
  it('sets transferSearchInput', () => {
    expect(
      reducer(Map(), action.setTransferSearchInput('mockTransferSearchInput'))
    ).toMatchSnapshot();
  });
  it('sets transferTabIndex', () => {
    expect(reducer(Map(), action.setTransferTabIndex(1))).toMatchSnapshot();
  });
  it('sets focusedTransferItemIndex', () => {
    expect(
      reducer(Map(), action.setFocusedTransferItemIndex(0))
    ).toMatchSnapshot();
  });
  it('sets userAssigned voice interaction TransferLists', () => {
    expect(
      reducer(
        Map(),
        action.setUserAssignedTransferLists(
          [{ mockTransferListKey: 'mockTransferListValue' }],
          'voice'
        )
      )
    ).toMatchSnapshot();
  });
  it('sets userAssigned non-voice interaction TransferLists', () => {
    expect(
      reducer(
        Map(),
        action.setUserAssignedTransferLists(
          [{ mockTransferListKey: 'mockTransferListValue' }],
          'nonVoice'
        )
      )
    ).toMatchSnapshot();
  });
  it('sets userAssigned voice interaction TransferLists LoadingState', () => {
    expect(
      reducer(
        Map(),
        action.setUserAssignedTransferListsLoadingState('voice', true)
      )
    ).toMatchSnapshot();
  });
  it('sets userAssigned non-voice interaction TransferLists LoadingState', () => {
    expect(
      reducer(
        Map(),
        action.setUserAssignedTransferListsLoadingState('nonVoice', true)
      )
    ).toMatchSnapshot();
  });
  it('sets userAssignedTransferListsVisibleState', () => {
    expect(
      reducer(
        Map(),
        action.setUserAssignedTransferListsVisibleState({
          'assignedTransferList-abc': true,
        })
      )
    ).toMatchSnapshot();
  });
  it('sets visibleStateOfAllUserAssignedTransferLists', () => {
    expect(
      reducer(Map(), action.setVisibleStateOfAllUserAssignedTransferLists(true))
    ).toMatchSnapshot();
  });
});
