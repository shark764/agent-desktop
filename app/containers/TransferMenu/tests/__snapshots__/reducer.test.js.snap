// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransferMenu reducer tests returns initial state 1`] = `
Immutable.Map {
  "transferSearchInput": "",
  "transferTabIndex": 0,
  "focusedTransferItemIndex": -1,
  "showTransferDialpad": false,
  "userAssignedTransferLists": Immutable.Map {
    "voiceListsLoadingState": true,
    "nonVoiceListsLoadingState": true,
  },
}
`;

exports[`TransferMenu reducer tests sets agentsListVisibleState 1`] = `
Immutable.Map {
  "agentsListVisibleState": true,
}
`;

exports[`TransferMenu reducer tests sets focusedTransferItemIndex 1`] = `
Immutable.Map {
  "focusedTransferItemIndex": 0,
}
`;

exports[`TransferMenu reducer tests sets queuesListVisibleState 1`] = `
Immutable.Map {
  "queuesListVisibleState": true,
}
`;

exports[`TransferMenu reducer tests sets transferSearchInput 1`] = `
Immutable.Map {
  "transferSearchInput": "mockTransferSearchInput",
}
`;

exports[`TransferMenu reducer tests sets transferTabIndex 1`] = `
Immutable.Map {
  "transferTabIndex": 1,
}
`;

exports[`TransferMenu reducer tests sets userAssigned non-voice interaction TransferLists 1`] = `
Immutable.Map {
  "userAssignedTransferLists": Immutable.Map {
    "nonVoiceListsLoadingState": false,
    "onlyQueueTransferLists": Immutable.List [
      Immutable.Map {
        "mockTransferListKey": "mockTransferListValue",
      },
    ],
  },
}
`;

exports[`TransferMenu reducer tests sets userAssigned non-voice interaction TransferLists LoadingState 1`] = `
Immutable.Map {
  "userAssignedTransferLists": Immutable.Map {
    "nonVoiceListsLoadingState": true,
  },
}
`;

exports[`TransferMenu reducer tests sets userAssigned voice interaction TransferLists 1`] = `
Immutable.Map {
  "userAssignedTransferLists": Immutable.Map {
    "voiceListsLoadingState": false,
    "pstnSipQueueTransferLists": Immutable.List [
      Immutable.Map {
        "mockTransferListKey": "mockTransferListValue",
      },
    ],
  },
}
`;

exports[`TransferMenu reducer tests sets userAssigned voice interaction TransferLists LoadingState 1`] = `
Immutable.Map {
  "userAssignedTransferLists": Immutable.Map {
    "voiceListsLoadingState": true,
  },
}
`;

exports[`TransferMenu reducer tests sets userAssignedTransferListsVisibleState 1`] = `
Immutable.Map {
  "userAssignedTransferLists": Immutable.Map {
    "transferListsVisibleState": Immutable.Map {
      "assignedTransferList-abc": true,
    },
  },
}
`;

exports[`TransferMenu reducer tests sets visibleStateOfAllUserAssignedTransferLists 1`] = `
Immutable.Map {
  "userAssignedTransferLists": Immutable.Map {
    "allTransferListsVisibleState": true,
  },
}
`;

exports[`TransferMenu reducer tests updates showTransferDialpad state 1`] = `
Immutable.Map {
  "showTransferDialpad": true,
}
`;
